### PR TITLE
CI: switch GPU pool

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

using our internal pool

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki, any thoughts on why these tests are failing on RTX 3090? :otter:

**Comment by SkafteNicki** [comment]
@Borda tests in LTS are failing with 
```ImportError: cannot import name 'CommitOperationAdd' from 'huggingface_hub'```
so maybe has to do v0.10.0 release of huggingface hub released 10 days ago.

However, not sure how this is related to changing gpu config...

**Comment by Borda** [comment]
Well, I fixed the testing as `pandas` reinstall was needed and lowered some other tol, but I feel it is not the very right way as it may bring some instability/correctness... @SkafteNicki

**Comment by SkafteNicki** [comment]
@Borda what hardware was we using before?

**Comment by Borda** [comment]
> @Borda what hardware was we using before?

before we used V100, and even before P100, and in fact I think there were some problems with both HW transitions... :(

**Comment by SkafteNicki** [comment]
It makes no sense to me why the tests are failing due to tolerance. Like theses:
```bash
FAILED unittests/pairwise/test_pairwise_distance.py::TestPairwise::test_pairwise_functional[sum-pairwise_cosine_similarity-cosine_similarity-x0-y0]
FAILED unittests/pairwise/test_pairwise_distance.py::TestPairwise::test_pairwise_functional[sum-pairwise_cosine_similarity-cosine_similarity-x1-y1]
FAILED unittests/pairwise/test_pairwise_distance.py::TestPairwise::test_pairwise_functional[sum-pairwise_euclidean_distance-euclidean_distances-x0-y0]
FAILED unittests/pairwise/test_pairwise_distance.py::TestPairwise::test_pairwise_functional[sum-pairwise_euclidean_distance-euclidean_distances-x1-y1]
FAILED unittests/pairwise/test_pairwise_distance.py::TestPairwise::test_pairwise_functional[sum-pairwise_linear_similarity-linear_kernel-x0-y0]
FAILED unittests/pairwise/test_pairwise_distance.py::TestPairwise::test_pairwise_functional[sum-pairwise_linear_similarity-linear_kernel-x1-y1]
```
because they succeed when the tests run where the first argument is `mean` and not `sum`. I cannot understand how it comes down to the scores being normalized or not if the test fail/pass.

**Comment by Borda** [comment]
> It makes no sense to me why the tests are failing due to tolerance. Like theses:

I would be curious if the reference changed or ours...

**Comment by SkafteNicki** [comment]
> > It makes no sense to me why the tests are failing due to tolerance. Like theses:
> 
> I would be curious if the reference changed or ours...

If the reference changed we should see the same tests failing for other PRs, so it must be our implementation something happen to

**Comment by Borda** [comment]
Thinking if we have some systematic issue here if on another GPU it does not give the same results... :/

**Comment by Borda** [comment]
Setting this as a priority as it is very strange that some tests are failing just because of changed HW - graphic card... :/

File: tests/unittests/image/test_lpips.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def _compare_fn(img1: Tensor, img2: Tensor, net_type: str, normalize: bool, redu
 
 @pytest.mark.skipif(not _LPIPS_AVAILABLE, reason="test requires that lpips is installed")
 class TestLPIPS(MetricTester):
-    atol: float = 1e-6
+    atol: float = 1e-3
```
Comment (Borda) [inline tests/unittests/image/test_lpips.py:48]:
```suggestion
    atol: float = 1e-6
```

File: tests/unittests/pairwise/test_pairwise_distance.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def _sk_metric(x, y, sk_fn, reduction):
 class TestPairwise(MetricTester):
     """test pairwise implementations."""
 
-    atol = 1e-4
+    atol = 1e-3
```
Comment (Borda) [inline tests/unittests/pairwise/test_pairwise_distance.py:81]:
```suggestion
    atol = 1e-4
```

File: tests/unittests/text/test_bertscore.py [hunk 3]
```diff
@@ -41,7 +41,7 @@
 MODEL_NAME = "albert-base-v2"
 
 
-def _assert_list(preds: Any, targets: Any, threshold: float = 1e-8):
+def _assert_list(preds: Any, targets: Any, threshold: float = 1e-6):
```
Comment (Borda) [inline tests/unittests/text/test_bertscore.py:44]:
```suggestion
def _assert_list(preds: Any, targets: Any, threshold: float = 1e-7):
```

File: tests/unittests/text/test_infolm.py [hunk 4]
```diff
@@ -100,7 +100,7 @@ def reference_infolm_score(preds, target, model_name, information_measure, idf,
 @pytest.mark.skipif(not _TRANSFORMERS_AVAILABLE, reason="test requires transformers")
 class TestInfoLM(TextTester):
     # Set atol = 1e-4 as reference results are rounded
-    atol = 1e-4
+    atol = 1e-3
```
Comment (Borda) [inline tests/unittests/text/test_infolm.py:103]:
```suggestion
    atol = 1e-4
```

File: .azure/gpu-pipeline.yml [hunk 5]
```diff
@@ -70,11 +78,13 @@ jobs:
         sudo apt-get install -y build-essential gcc g++ cmake ffmpeg git libsndfile1 unzip --no-install-recommends
         # python -m pip install "pip==20.1"
         pip install -e . --requirement ./requirements/devel.txt
+        pip install pandas --force-reinstall
         pip install mkl-service==2.4.0  # needed for the gpu multiprocessing
         pip list
       displayName: 'Install environment'
 
     - bash: |
+        python -c "from torch import __version__ as ver ; assert str(ver) == '$(torch-ver)', f'PyTorch: {ver}'"
```
Comment (Borda) [inline .azure/gpu-pipeline.yml:87]: @SkafteNicki turned out what we were always assumed to be testing LTS (1.8.1) was, in fact always reinstalled with the latest/stable :see_no_evil:

**Comment by Borda** [comment]
> > @Borda what hardware was we using before?
> 
> before we used V100, and even before P100, and in fact I think there were some problems with both HW transitions... :(

correction:
- LTS (v1.8.1) uses K80
- stable (1.13.0) uses P100

**Review by ethanwharris — APPROVED** [review]

**Pull request merged**